controller/src/beerocks/master/beerocks_master_main.cpp

```diff
@@ -344,6 +344,17 @@ int main(int argc, char *argv[])
     son::db master_db(master_conf, logger, bridge_info.mac);
     LOG(DEBUG) << "slave_keep_alive_retries=" << master_db.config.slave_keep_alive_retries;
     // diagnostics_thread diagnostics(master_db);
+
+    LOG(DEBUG) << "persistent db enable=" << master_conf.persistent_db;
+    if (master_conf.persistent_db) {
+        LOG(DEBUG) << "loading clients from persistent db";
+        if (!master_db.load_persistent_db_clients()) {
+            LOG(ERROR) << "failed to load clients from persistent db";
+        } else {
+            LOG(DEBUG) << "load clients from persistent db finished successfully";
+        }
+    }
```

Comment:
Why is it here and not on `son_master_thread::init()`?

Reply:
same as why the `master_db` is created before the `son_master_thread:init()` - I consider this operation as pre-init preparations.

Reply:
The database was created here originally because it was planned to be a shared object between several threads on the process, not because it is a "pre-init" operation.
That's why the database lock/unlock feature was implemented.
I believe that the `main` files should be with less code as possible, and if there is not a really good reason to put it here, then it should be on the `son_master::init()`

Reply:
I will move in inside the `son_master::init()`